src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -525,29 +527,67 @@ export function useCourseEnrollmentsBySection(courseEnrollmentsByStatus) {
   };
 }
 
+function handleQueriesForUpdatedCourseEnrollmentStatus({
+  queryClient,
+  enterpriseSlug,
+  enterpriseCustomer,
+  courseRunId,
+  updatedEnrollmentParams,
+}) {
+  // Transformation
+  const { newStatus } = updatedEnrollmentParams;
+  const transformUpdatedEnrollment = (enrollment) => {
+    if (enrollment.courseRunId !== courseRunId) {
+      return enrollment;
+    }
+    return {
+      ...enrollment,
+      courseRunStatus: newStatus,
+    };
+  };
+  const isBFFEnabled = isBFFEnabledForEnterpriseCustomer(enterpriseCustomer.uuid);
+
+  if (isBFFEnabled) {
+  // Determine which BFF queries need to be updated after unenrolling.
+    const dashboardBFFQueryKey = queryEnterpriseLearnerDashboardBFF(
+      { enterpriseSlug },
+    ).queryKey;
+    const bffQueryKeysToUpdate = [dashboardBFFQueryKey];
+    // Update the enterpriseCourseEnrollments data in the cache for each BFF query.
+    bffQueryKeysToUpdate.forEach((queryKey) => {
+      const existingBFFData = queryClient.getQueryData(queryKey);
```

Comment:
Similar comment here around having a conditional check for only calling `setQueryData` when there's already-cached data for the query key.